toolchain/lowering/lowering_context.cpp

```diff
@@ -144,6 +144,12 @@ auto LoweringContext::BuildType(SemanticsNodeId node_id) -> llvm::Type* {
 
   auto node = semantics_ir_->GetNode(node_id);
   switch (node.kind()) {
+    case SemanticsNodeKind::ArrayType: {
+      auto [bound_node_id, type_id] = node.GetAsArrayType();
+      auto bound_value = semantics_ir_->GetArrayBoundValue(bound_node_id);
+      llvm::SmallVector<llvm::Type*> subtypes(bound_value, GetType(type_id));
+      return llvm::StructType::get(*llvm_context_, subtypes);
```

Comment:
LLVM provides an ArrayType which should be suitable for use here.

Reply:
As we are using the tuple value as the array value, I used StructType to match with the tuple type. Should I change it to ArrayType?

Reply:
Yes, that's correct.